Source/WebCore/Modules/compression/CompressionStreamEncoder.h

```diff
@@ -69,10 +71,9 @@ class CompressionStreamEncoder : public RefCounted<CompressionStreamEncoder> {
     const size_t startingAllocationSize = 16384; // 16KB
     const size_t maxAllocationSize = 1073741824; // 1GB
 
-    bool m_initialized { false };
     bool m_didFinish { false };
-    z_stream m_zstream;
-
-    Formats::CompressionFormat m_format;
+    ZStream m_zstream;
+    CompressionStream m_compressionStream;
```

Comment:
Conceptually, this are mutually exclusive, right?
Maybe we can use `variant`? It's not a big deal to land this as is though.

Reply:
Added `TODO` for work in a follow up patch.